src/TriggerBinding/SqlTableChangeMonitor.cs

```diff
@@ -85,6 +85,12 @@ internal sealed class SqlTableChangeMonitor<T> : IDisposable
         private int _leaseRenewalCount = 0;
         private State _state = State.CheckingForChanges;
 
+        /// <summary>
+        /// The result from the last time we triggered the user function. This will normally be cleared immediately after executing
+        /// the function but if an error occurs during cleanup we want to keep retrying the cleanup until it succeeds.
+        /// </summary>
+        private FunctionResult _triggeredFunctionResult = null;
```

Comment:
I'm trying to rationalize this with the State enum. Have you considered adding an explicit new item in that enum instead of saving the FunctionResult and checking against null?

Reply:
Alright, tried cleaning it up a bit. Now there's 3 states and I removed the need for tracking the success of the triggered function. Instead, we now have a separate list for tracking what rows are marked to have their leases released. This makes it clearer what the purpose of the variable is and makes it easier to ensure that we only release leases for rows that are done being processed.